### PR TITLE
Added tests for lesson 10

### PR DESCRIPTION
**Pull request by Pavellys**
Added chromedriver.exe
Added new tests

### CODE REVIEW

File: src/test/java/pages/ContextMenuPage.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        super(driver);
+    }
+
+    Actions actions = new Actions(driver);
+    public static final String CONTEXT_MENU_URL = "http://the-internet.herokuapp.com/context_menu";
+    public static final By ELEMENT_RIGHT_CLICK = By.xpath("//*[@id='hot-spot']");
+    public final String EXP_RESULT_ALERT = "You selected a context menu";
+    /*Open website
+    http://the-internet.herokuapp.com/context_menu
+     */
```
Comment (vchesnakoutest) [inline src/test/java/pages/ContextMenuPage.java]: format should be     
     /**
     * Open website http://the-internet.herokuapp.com/context_menu
     */

please fix for others as well

File: src/test/java/pages/ContextMenuPage.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import org.openqa.selenium.*;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+public class ContextMenuPage extends BasePage {
+    public ContextMenuPage(WebDriver driver) {
+        super(driver);
+    }
+
+    Actions actions = new Actions(driver);
```
Comment (vchesnakoutest) [inline src/test/java/pages/ContextMenuPage.java]: move creation of Actions object to methods where it used

File: src/test/java/pages/ContextMenuPage.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    }
+    /*
+    Find element for test and context click
+     */
+    public void rightClickToElement() {
+        WebElement element = driver.findElement(ELEMENT_RIGHT_CLICK);
+        actions.contextClick(element).perform();
+        wait.until(ExpectedConditions.alertIsPresent());
+    }
+    /*
```
Comment (vchesnakoutest) [inline src/test/java/pages/ContextMenuPage.java]: 1 empty line before method declaration

File: src/test/java/pages/ContextMenuPage.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    Checking the warning after a click
+     */
+    public String isAlert(){
```
Comment (vchesnakoutest) [inline src/test/java/pages/ContextMenuPage.java]: prefix 'is' should be used for boolean methods, use

getAlertText()

File: src/test/java/pages/ContextMenuPage.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        Alert alert = driver.switchTo().alert();
+        String alertText = alert.getText();
+        return alertText;
```
Comment (vchesnakoutest) [inline src/test/java/pages/ContextMenuPage.java]: you can 
`return alert.getText();`

minus 1 line of code

File: src/test/java/pages/DynamicControlsPage.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+
+public class DynamicControlsPage extends BasePage {
+    public static final String DYNAMIC_CONTROLS_URL = "http://the-internet.herokuapp.com/dynamic_controls";
+    public final By CHECKBOX_ELEMENT = By.id("checkbox");
+    public final By BUTTON_ENABLE = By.xpath("/html/body/div[2]/div/div[1]/form[2]/button");
```
Comment (vchesnakoutest) [inline src/test/java/pages/DynamicControlsPage.java]: bad locator, use more smart xpath. using text() for example

File: src/test/java/pages/DynamicControlsPage.java [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    public final By FIELD_INPUT = By.xpath("//*[@id='input-example']/input");
+    public final By BUTTON_REMOVE = By.xpath("//*[@id='checkbox-example']/button");
+    public final By TEXT_AFTER_CLICK_ENABLE = By.xpath("/html/body/div[2]/div/div[1]/form[2]/p");
```
Comment (vchesnakoutest) [inline src/test/java/pages/DynamicControlsPage.java]: bad locator, use more smart xpath

File: src/test/java/pages/DynamicControlsPage.java [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    public final By TEXT_AFTER_BUTTON = By.id("message");
+    public final boolean EXP_RESULT_CHECKBOX = false;
+    public final boolean EXP_RESULT_INPUTFIELD = true;
+    public final String EXP_RESULT_MESSAGE_CHECKBOX = "It's gone!";
+    public final String EXP_RESULT_MESSAGE_INPUTFIELD = "It's enabled!";
```
Comment (vchesnakoutest) [inline src/test/java/pages/DynamicControlsPage.java]: 1 ) expected results should not be declared in Page classes, they are considered as Test data
2 ) not necessary to declare constants for boolean values

File: src/test/java/pages/DynamicControlsPage.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+     */
+    @Override
+    public void openPage() {
+        driver.get(DYNAMIC_CONTROLS_URL);
+
+    }
+    /*
+    Checking checkbox
+     */
+    public boolean isCheckbox() {
```
Comment (vchesnakoutest) [inline src/test/java/pages/DynamicControlsPage.java]: `isCheckboxDisplayed()`

File: src/test/java/pages/DynamicControlsPage.java [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+            driver.findElement(CHECKBOX_ELEMENT).isDisplayed();
+            return true;
+        } catch (NoSuchElementException e) {
+            return false;
+        }
+    }
+    /*
+    Find element on the path
+     */
+    public WebElement findElementForTest(By element) {
```
Comment (vchesnakoutest) [inline src/test/java/pages/DynamicControlsPage.java]: bad approach, don't pass locators in Test classes like you did. Make separate methods for click on different locators.

File: src/test/java/pages/DynamicControlsPage.java [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+    /*
+    Waits until the element disappears
+     */
+    public void waiterCheckbox() {
+        wait.until(ExpectedConditions.invisibilityOfElementLocated(CHECKBOX_ELEMENT));
+    }
+    /*
+    Waits until the element to appears
+     */
+    public void waiterEnable(){
```
Comment (vchesnakoutest) [inline src/test/java/pages/DynamicControlsPage.java]: `waitForCheckboxDisplayed()` and so on....

File: src/test/java/pages/DynamicControlsPage.java [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+
+
+public class DynamicControlsPage extends BasePage {
+    public static final String DYNAMIC_CONTROLS_URL = "http://the-internet.herokuapp.com/dynamic_controls";
+    public final By CHECKBOX_ELEMENT = By.id("checkbox");
+    public final By BUTTON_ENABLE = By.xpath("/html/body/div[2]/div/div[1]/form[2]/button");
+    public final By FIELD_INPUT = By.xpath("//*[@id='input-example']/input");
+    public final By BUTTON_REMOVE = By.xpath("//*[@id='checkbox-example']/button");
+    public final By TEXT_AFTER_CLICK_ENABLE = By.xpath("/html/body/div[2]/div/div[1]/form[2]/p");
+    public final By TEXT_AFTER_BUTTON = By.id("message");
```
Comment (vchesnakoutest) [inline src/test/java/pages/DynamicControlsPage.java]: name should be REMOVE_RESULT_TEXT or smth like that

File: src/test/java/pages/DynamicControlsPage.java [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+    /*
+    Waits until the element to appears
+     */
+    public void waiterEnable(){
+        wait.until(ExpectedConditions.visibilityOfElementLocated(TEXT_AFTER_BUTTON));
+    }
+    /*
+    Get text after click to button "Remove"
+     */
+    public String getTextAfterRemove() {
```
Comment (vchesnakoutest) [inline src/test/java/pages/DynamicControlsPage.java]: `getRemoveResultText()`

File: src/test/java/pages/DynamicControlsPage.java [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+    /*
+    Get text after click to button "Enable"
+     */
+    public String getTextAfterEnable() {
+        return driver.findElement(TEXT_AFTER_CLICK_ENABLE).getText();
+    }
+    /*
+    Checking input field
+     */
+    public boolean isEnableInputField(){
```
Comment (vchesnakoutest) [inline src/test/java/pages/DynamicControlsPage.java]: `isInputFieldEnabled()`

File: src/test/java/pages/FileUploadPage.java [hunk 15]
```diff
@@ -0,0 +1,59 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+public class FileUploadPage extends BasePage{
+    public final String FILE_UPLOAD_URL = "http://the-internet.herokuapp.com/upload";
+    public final String FILE_PATH = "C:\\Users\\dkwr2\\OneDrive\\Desktop\\test.txt";
```
Comment (vchesnakoutest) [inline src/test/java/pages/FileUploadPage.java:9]: FILE_PATH should not be in Page class, it's Test data

File: src/test/java/pages/FileUploadPage.java [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+    public final By UPLOAD_PATH = By.id("file-upload");
+    public final By BUTTON_UPLOAD = By.id("file-submit");
+    public final By PATH_UPLOAD_DONE = By.id("uploaded-files");
+    public static final String ERROR = "error";
+    public final String EXP_FILE_NAME = "test.txt";
```
Comment (vchesnakoutest) [inline src/test/java/pages/FileUploadPage.java:14]: this is test data

File: src/test/java/pages/FramesPage.java [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+
+import java.awt.*;
+
+public class FramesPage extends BasePage{
+    public static final String FRAMES_URL = "http://the-internet.herokuapp.com/frames";
+    public static final By PATH_ELEMENT_IFRAME = By.xpath("//*[@href='/iframe']");
+    public static final By PATH_TEXT_IN_FRAME = By.id("tinymce");
+    public final String EXP_TEXT_FROM_FRAME = "Your content goes here.";
```
Comment (vchesnakoutest) [inline src/test/java/pages/FramesPage.java:12]: this is test data

File: src/test/java/pages/FileUploadPage.java [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+    Open website http://the-internet.herokuapp.com/upload
+     */
+    @Override
+    public void openPage() {
+        driver.get(FILE_UPLOAD_URL);
+    }
+    /*
+    Enter path to file
+     */
+    public void uploadFilePath(){
```
Comment (vchesnakoutest) [inline src/test/java/pages/FileUploadPage.java]: just uploadFile and pass file path as parameter in test

**Review by vchesnakoutest — COMMENTED** [review]
Please make fixes according to comments

File: src/test/java/pages/ContextMenuPage.java (hunk 2 above)
Comment (Pavellys) [reply src/test/java/pages/ContextMenuPage.java]: done

File: src/test/java/pages/ContextMenuPage.java (hunk 1 above)
Comment (Pavellys) [reply src/test/java/pages/ContextMenuPage.java]: done

File: src/test/java/pages/ContextMenuPage.java (hunk 3 above)
Comment (Pavellys) [reply src/test/java/pages/ContextMenuPage.java]: done

File: src/test/java/pages/ContextMenuPage.java (hunk 4 above)
Comment (Pavellys) [reply src/test/java/pages/ContextMenuPage.java]: done

File: src/test/java/pages/ContextMenuPage.java (hunk 5 above)
Comment (Pavellys) [reply src/test/java/pages/ContextMenuPage.java]: done

File: src/test/java/pages/DynamicControlsPage.java (hunk 6 above)
Comment (Pavellys) [reply src/test/java/pages/DynamicControlsPage.java]: done

File: src/test/java/pages/DynamicControlsPage.java (hunk 7 above)
Comment (Pavellys) [reply src/test/java/pages/DynamicControlsPage.java]: done

File: src/test/java/pages/DynamicControlsPage.java (hunk 12 above)
Comment (Pavellys) [reply src/test/java/pages/DynamicControlsPage.java]: done

File: src/test/java/pages/DynamicControlsPage.java (hunk 8 above)
Comment (Pavellys) [reply src/test/java/pages/DynamicControlsPage.java]: done

File: src/test/java/pages/DynamicControlsPage.java (hunk 9 above)
Comment (Pavellys) [reply src/test/java/pages/DynamicControlsPage.java]: done

File: src/test/java/pages/DynamicControlsPage.java (hunk 10 above)
Comment (Pavellys) [reply src/test/java/pages/DynamicControlsPage.java]: done

File: src/test/java/pages/DynamicControlsPage.java (hunk 13 above)
Comment (Pavellys) [reply src/test/java/pages/DynamicControlsPage.java]: done

File: src/test/java/pages/DynamicControlsPage.java (hunk 11 above)
Comment (Pavellys) [reply src/test/java/pages/DynamicControlsPage.java]: done

File: src/test/java/pages/DynamicControlsPage.java (hunk 14 above)
Comment (Pavellys) [reply src/test/java/pages/DynamicControlsPage.java]: done

File: src/test/java/pages/FileUploadPage.java (hunk 15 above)
Comment (Pavellys) [reply src/test/java/pages/FileUploadPage.java:9]: done

File: src/test/java/pages/FileUploadPage.java (hunk 16 above)
Comment (Pavellys) [reply src/test/java/pages/FileUploadPage.java:14]: done

File: src/test/java/pages/FileUploadPage.java (hunk 18 above)
Comment (Pavellys) [reply src/test/java/pages/FileUploadPage.java]: done

File: src/test/java/pages/FramesPage.java (hunk 17 above)
Comment (Pavellys) [reply src/test/java/pages/FramesPage.java:12]: done

File: src/test/java/pages/ContextMenuPage.java (hunk 1 above)
Comment (Pavellys) [reply src/test/java/pages/ContextMenuPage.java]: done

**Review by vchesnakoutest — APPROVED** [review]
good work, please merge 👍

**Pull request merged**